File: 2 Assignment 3/UltimatePlayer.java
/*
 * Ryan Lutz
 * 10/14/19
 * M2A3
 */
public class UltimatePlayer extends Person {
    private int jerseyNumber; // Declare vars
    private String position;
    private static int num;

    public UltimatePlayer(String fName, String lName, String pos) { // Constructor
        super(fName, lName); // Call Person constructor
        position = "handler"; // Set vars to appropriate vals
        if (pos == "cutter") {
            position = "cutter";
        }
        num++;
        jerseyNumber = num;
    }

    public String getPosition() { // Get player's position
        return position; // Return position
    }

    public String toString() { // To String
        return super.toString() + "\n   Jersey #: " + jerseyNumber + "\n   Position: " + position; // Return formatted string
    }
}

File: 2 Assignment 3 kt/UltimatePlayer.kt
open class UltimatePlayer:Person {
  private companion object {
    private var num:Int = 0
    fun getNum():Int {return num}
    fun addNum() {num++}
  }

  private var jerseyNumber:Int
  private var position:String

  constructor(firstName:String, lastName:String, pos:String):super(firstName, lastName) {
    position = "handler"
    if (pos.equals("cutter")) {
      position = "cutter"
    }
    addNum()
    jerseyNumber = getNum()
  }

  fun getPosition():String {
    return position
  }

  override fun toString():String {
    return "${super.toString()}\n   Jersey #: $jerseyNumber\n   Position: $position"
  }
}